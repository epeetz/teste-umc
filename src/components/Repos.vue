<template>
	<div class="container mb-3">
		<div class="d-flex mb-3">
			<div class="mr-auto">
				<h3>Repositories by {{ reformattedSearchString }}</h3>
			</div>
		</div>
		<div class="card-columns">
			<div class="card" v-bind:key="repo.id" v-for="repo in repos">
				<div>
					<div class="card-body">
						<h5 class="card-title">
							<a
							:href="repo.html_url"
							class="card-link"
							target="_blank">{{ repo.name }}</a>
						</h5>
						<h6 class="card-subtitle mb-2 text-muted" >{{ repo.created_at | formatDate }}</h6>
						<p class="card-text">{{ repo.description }}</p>
					</div>
				</div>
			</div>
		</div>
	</div>
</template>

<script>

export default {
	name: 'Repos',
	props: ['repos','reformattedSearchString']
};
</script>

<style scoped>
button:focus {
	box-shadow: none !important;
}
</style>