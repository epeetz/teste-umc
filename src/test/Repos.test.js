import { shallowMount } from '@vue/test-utils';
import Repos from '../components/Repos';

let wrapper;

beforeEach(() => {
    wrapper = shallowMount(Repos, {
        propsData: {},
        mocks: {},
        stubs: {},
        methods: {},
    });
});

afterEach(() => {
    wrapper.destroy();
});

describe('Repos', () => {
    test('is a Vue instance', () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });
    test('rendering', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});