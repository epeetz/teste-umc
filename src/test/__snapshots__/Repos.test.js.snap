// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos rendering 1`] = `"<div class=\\"container mb-3\\"><div class=\\"d-flex mb-3\\"><div class=\\"mr-auto\\"><h3>Repositories by </h3></div></div> <div class=\\"card-columns\\"></div></div>"`;
